src/admin/class-wordlift-admin-entity-list.php

```diff
@@ -80,4 +80,81 @@ public function render_custom_columns( $column, $entity_id ) {
 		}
 		
 	}
+	
+	/**
+	 * Add wl-classification-scope select box before the 'Filter' button.
+	 *
+	 * @since 3.3.0
+	 */
+	public function restrict_manage_posts_classification_scope() {
+		
+		// Only show on entity list page
+		$screen = get_current_screen();
+		if( $screen->post_type !== Wordlift_Entity_Service::TYPE_NAME ){
+			return;
+		}
+		
+		// Was a W already selected?
+		$selected = isset( $_GET['wl-classification-scope'] ) ? $_GET['wl-classification-scope'] : '' ;
+		
+		// Print select box with the 4W
+		$all_w = array(
+			"All 'W'",
+			WL_WHAT_RELATION,
+			WL_WHO_RELATION,
+			WL_WHERE_RELATION,
+			WL_WHEN_RELATION
+		);
+		echo '<select name="wl-classification-scope" id="wl-dropdown-classification-scope">';
+		foreach ( $all_w as $w ) {
+			$default  = ( $selected === $w ) ? 'selected' : '';
+			echo sprintf( '<option value="%s" %s >%s</option>', $w, $default, $w );
+		}
+		echo '</select>';
+	}
+
+	/**
+	 * Server side response operations for the classification filter set in *restrict_manage_posts_classification_scope_filter*
+	 *
+	 * @since 3.3.0
+	 * 
+	 * @param array $clauses WP main query clauses.
+	 * 
+	 * @return array Modified clauses.
+	 */
+	public function posts_clauses_classification_scope( $clauses ) {
+		
+		// Run only on admin page
+		if( ! is_admin() ) {
+			return $clauses;
+		}
+		
+		// Only apply on entity list page, only if this is the main query and if the wl-classification-scope query param is set
+		$screen = get_current_screen();
+		if( ! ( $screen->post_type === Wordlift_Entity_Service::TYPE_NAME && is_main_query() && isset( $_GET['wl-classification-scope'] ) ) ) {
```

Comment:
Pass the Entity Service trough the constructor ... Then you have the method available